### PR TITLE
Collapsed PanelHeader adjustments

### PR DESCRIPTION
**Pull request by TomTirapani**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/panel/impl/PanelHeader.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const panelHeader = hoistCmp.factory({
         const panelModel = useContextModel(PanelModel),
             {collapsed, collapsible, isModal, vertical, side} = panelModel,
             {title, icon, compact} = props,
+            collapsedTitle = props.collapsedTitle ?? title,
```
Comment (lbwexler) [inline desktop/cmp/panel/impl/PanelHeader.ts:25]: think we should use withDefault() to allow `null`  to be respected

File: desktop/cmp/panel/impl/PanelHeader.ts [hunk 2]
```diff
@@ -48,9 +49,7 @@ export const panelHeader = hoistCmp.factory({
                         ? box({
                               className: titleCls,
                               flex: 1,
-                              item: isString(title)
-                                  ? span({className: `${titleCls}__inner`, item: title})
-                                  : title
+                              item: span({className: `${titleCls}__inner`, item: title})
```
Comment (ghsolomon) [inline desktop/cmp/panel/impl/PanelHeader.ts:52]: Think this needs to be `collapsed ? collapsedTitle : title` for the vertical collapsed case

**Review by ghsolomon — CHANGES_REQUESTED** [review]

**Comment by TomTirapani** [comment]
Thanks for the comments guys - PR updated 👍

File: desktop/cmp/panel/impl/PanelHeader.ts [hunk 3]
```diff
@@ -22,7 +23,8 @@ export const panelHeader = hoistCmp.factory({
         const panelModel = useContextModel(PanelModel),
             {collapsed, collapsible, isModal, vertical, side} = panelModel,
             {title, icon, compact} = props,
-            collapsedTitle = props.collapsedTitle ?? title,
+            collapsedTitle = withDefault(props.collapsedTitle, title),
+            displayedTitle = collapsed ? collapsedTitle : title,
```
Comment (ghsolomon) [inline desktop/cmp/panel/impl/PanelHeader.ts:27]: Nice!

**Review by ghsolomon — APPROVED** [review]

**Pull request merged**